File: ivef-lib/trunk/ivef-objc/Classes/IVEFPos.m

#import "IVEFPos.h"


@implementation IVEFPos

- (id) init {
    self = [super init];
    if (self != nil) {
        m_dateFormatter = [[NSDateFormatter alloc] init];
        [m_dateFormatter setDateFormat:@"yyyy-MM-dd'T'HH:mm:ss.SSS"];
    }
    return self;
}

-(void) setLat:(float) val {

    if (val < -90)
        return;
    if (val > 90)
        return;
    m_lat = val;
}

- (float) getLat {

    return m_lat;
}

-(void) setLong:(float) val {

    if (val < -180)
        return;
    if (val > 180)
        return;
    m_long = val;
}

- (float) getLong {

    return m_long;
}

-(void) setAttributes:(NSDictionary *)attributeDict {

        for (NSString *key in attributeDict) {
            if ([key isEqualToString: @"Lat"]) {
                NSString *value = [attributeDict objectForKey: key];
                float val = [value floatValue];
                [self setLat: val];
            }
            else if ([key isEqualToString:@"Long"]) {
                NSString *value = [attributeDict objectForKey: key];
                float val = [value floatValue];
                [self setLong: val];
            }
        }
}

-(NSString *) XML {

    NSMutableString *xml = [NSMutableString stringWithString:@"<Pos"];
    [xml appendString: @" Lat=\""];
    [xml appendString: [NSString stringWithFormat:@"%f", m_lat]];
    [xml appendString: @"\""];
    [xml appendString: @" Long=\""];
    [xml appendString: [NSString stringWithFormat:@"%f", m_long]];
    [xml appendString: @"\""];
    [xml appendString:@">\n"];
    [xml appendString: @"</Pos>\n"];
    return xml;
}

-(NSString *) stringValueWithLead: (NSString *) lead {

    NSMutableString *str = [[[NSMutableString alloc] init] autorelease];
    [str setString: [lead stringByAppendingString:@"Pos\n"]];
    [str appendString: [lead stringByAppendingString: @" "]];
    [str appendString: @"Lat=\""];
    [str appendString: [NSString stringWithFormat:@"%f", m_lat]];
    [str appendString: @"\"\n"];

    [str appendString: [lead stringByAppendingString: @" "]];
    [str appendString: @"Long=\""];
    [str appendString: [NSString stringWithFormat:@"%f", m_long]];
    [str appendString: @"\"\n"];

    return str;
}


@end

File: OLD_tags/IVEF_0_1_0_RC2/ivef-qt/IVEFPos.cpp

#include "IVEFPos.h"

Pos::Pos() {

}

Pos::Pos(const Pos &val) : QObject() {

    m_lat = val.m_lat;
    m_long = val.m_long;
}

Pos & Pos::operator=(const Pos &val) {

    m_lat = val.m_lat;
    m_long = val.m_long;
    return *this;
}

void Pos::setLat(float val) {

    if (val < -90)
        return;
    if (val > 90)
        return;
    m_lat = val;
}

float Pos::getLat() const {

    return m_lat;
}

void Pos::setLong(float val) {

    if (val < -180)
        return;
    if (val > 180)
        return;
    m_long = val;
}

float Pos::getLong() const {

    return m_long;
}

QString Pos::toXML() {

    QString xml = "<Pos";
    xml.append(" Lat=\"" + QString::number(m_lat) + "\"");
    xml.append(" Long=\"" + QString::number(m_long) + "\"");
    xml.append(">\n");
    xml.append( "</Pos>\n");
    return xml;
}

QString Pos::toString(QString lead) {

    QString str = lead + "Pos\n";
    str.append( lead + "    Lat = " + QString::number(m_lat) + "\n");
    str.append( lead + "    Long = " + QString::number(m_long) + "\n");
    return str;
}

